contrib/http_wasm/filters/http/source/host/context.h

```diff
@@ -0,0 +1,277 @@
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <ctime>
+#include <functional>
+#include <iostream>
+#include <deque>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "contrib/http_wasm/filters/http/source/host/context_interface.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace HttpWasm {
+namespace Host {
+
+#include "contrib/http_wasm/filters/http/source/host/http_wasm_common.h"
+#include "contrib/http_wasm/filters/http/source/host/http_wasm_enums.h"
+class PluginHandleBase;
+class WasmBase;
+class WasmVm;
+
+/**
+ * PluginBase is container to hold plugin information which is shared with all Context(s) created
+ * for a given plugin. Embedders may extend this class with additional host-specific plugin
+ * information as required.
+ * @param name is the name of the plugin.
+ * @param root_id is an identifier for the in VM handlers for this plugin.
+ * @param vm_id is a string used to differentiate VMs with the same code and VM configuration.
+ * @param plugin_configuration is configuration for this plugin.
+ * @param fail_open if true the plugin will pass traffic as opposed to close all streams.
+ * @param key is used to uniquely identify this plugin instance.
+ */
+struct PluginBase {
```

Comment:
**Base structs, like PluginBase, ContextBase, are only existing to forcibly separate out Proxy-Wasm C++ and Envoy, so we won't need them eventually. We should end up having only `Plugin`/`Context`.

Reply:
done